docs/postgres.md

```diff
@@ -0,0 +1,110 @@
+# PostgreSQL
+
+Coder ships with a built-in PostgreSQL database, but if you'd like to set up and
+use your own, refer to the following instructions.
+
+For in-depth information, please see the [PostgreSQL documentation](https://www.postgresql.org/docs/current/tutorial-start.html).
+
+## PostgreSQL on macOS with Homebrew
+
+1. Install with Homebrew:
+
+    ```console
+    brew install postgres
+    ```
+
+1. Start PostgreSQL with brew
+
+    ```console
+    brew services start postgresql
+    ```
+
+1. Connect to PostgreSQL:
+
+    ```console
+    psql postgres
+    ```
+
+1. Create the `coderuser` role:
+
+    ```console
+    create role coder-user with login;
+    ```
+
+1. Create a database called `coder` and assign the owner:
+
+    ```console
+    create database coder owner coderuser;
+    ```
+
+1. Set the password for `coderuser`:
+
+    ```console
+    \password coder # enter password when prompted
+    ```
+
+1. Assign rights to the database to your user:
+
+    ```console
+    grant all privileges on database coder to coderuser;
+    ```
```

Comment:
Suggestion: we don't need to duplicate the database setup on different platforms, just installing + getting a postgres session open are different :-)